modules/installation/proc-configuring-an-rhdh-instance-with-tls-in-kubernetes.adoc

```diff
@@ -0,0 +1,104 @@
+* You have obtained the secret and the service CA certificate associated with your service account.
+* You have created some resources and added labels to them so they can be discovered by the Kubernetes plugin. For more information, see link:https://backstage.io/docs/features/kubernetes/configuration#common-backstageiokubernetes-id-label[Common `backstage.io/kubernetes-id` label].
+
+.Procedure
+
+. Enable the Kubernetes plugins in the `dynamic-plugins-rhdh.yaml` ConfigMap file:
++
+[source,yaml]
+----
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: dynamic-plugins-rhdh
+data:
+  dynamic-plugins.yaml: |
+    includes:
+      - dynamic-plugins.default.yaml
+    plugins:
+      - package: ./dynamic-plugins/dist/backstage-plugin-kubernetes-backend-dynamic
+        disabled: false <1>
+      - package: ./dynamic-plugins/dist/backstage-plugin-kubernetes
+        disabled: false <2>
+        # ...  
+----
+<1> Set the value to `false` to enable the `backstage-plugin-kubernetes-backend-dynamic` plugin
+<2> Set the value to `false` to enable the `backstage-plugin-kubernetes` plugin.
+
+. Set the kubernetes cluster details and configure the catalog sync options in the `app-config-rhdh.yaml` ConfigMap file:
++
+[source,yaml]
+----
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: app-config-rhdh
+data:
+  "app-config-rhdh.yaml": |
+  # ...
+  catalog:
+    rules:
+      - allow: [Component, System, API, Resource, Location]
+    providers:
+      kubernetes:
+        openshift:
+          cluster: openshift
+          processor:
+            namespaceOverride: default
+            defaultOwner: guests
+          schedule:
+            frequency:
+              seconds: 30
+            timeout:
+              seconds: 5
+  kubernetes:
+    serviceLocatorMethod:
+      type: 'multiTenant'
+    clusterLocatorMethods:
+      - type: 'config'
+        clusters:
+          - url: <target-cluster-api-server-url> <1>
+            name: openshift
+            authProvider: 'serviceAccount'
+            skipTLSVerify: false <2>
+            skipMetricsLookup: true
+            dashboardUrl: <target-cluster-console-url> <3>
+            dashboardApp: openshift
+            serviceAccountToken: <service-account-token> <4>
+            caData: <service-ca-base64-encoded> <5>
+            # ...
+----
+<1> The base URL to the Kubernetes control plane. You can run the `kubectl cluster-info` command to get the base URL.
+<2>  Set the value of this parameter to `false` to enable the verification of the TLS certificate.
+<3> Optional: The link to the Kubernetes dashboard managing the ARO cluster. 
+<4>  Optional: The service account token for using the `serviceAccount` authentication provider.
+<5> Set the value of the parameter to a base64-encoded certificate authority bundle in the .PEM format. 
```

Comment:
Should we probably suggest to use environment variables from a secret instead of directly putting sensitive data in the configMap?

Reply:
Rather than suggesting, can we include an extra step to let the user know that they can create a certificate scret and use it to configure the kuberneted cluster? I have made this change, could you please have a look? Thanks

Reply:
Is this supported? When I looked at the [backstage docs](https://github.com/backstage/backstage/blob/master/docs/features/kubernetes/configuration.md#clusterscadata-optional) my understanding was that the `caData` would require the base64-encoded certificate authority bundle in PEM format directly, not a reference to a secret. My suggestion was to simply do something like [this](https://github.com/backstage/backstage/blob/master/docs/features/kubernetes/configuration.md#configuring-kubernetes-clusters): specify `caData: ${K8S_CONFIG_CA_DATA}` where `K8S_CONFIG_CA_DATA` is just an environment variable defined in the usual rhdh-secret, like we suggest in other docs. Same for the other variables, like token and clusterName.

Reply:
Thank you for the clarification. 

Changed the description:
<5> Pass the CA data using a `K8S_CONFIG_CA_DATA` environment variable that you can define in your `secrets-rhdh` secret.

Reply:
Perfect. I would use the same approach for the others `<target-cluster-api-server-url>` `<target-cluster-console-url>` `<service-account-token>` variables, or the token at least.

Reply:
Made this change and added a variable only for the service account token because it contains sensitive data. I think we can leave the part of defining a variable for rest of the paramters to our readers.